File: nav_menu/app/src/main/java/com/codehafeez/nav_menu/MainActivity.java
package com.codehafeez.nav_menu;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.my_menu, menu);
//        menu.add(1, 1, 0, "Item1");
//        menu.add(1, 2, 1, "Item2");
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        Toast.makeText(MainActivity.this, "Menu Selected : "+item, Toast.LENGTH_SHORT).show();
        return true;
    }
}

File: ucrop-cropImageExample/app/src/main/java/com/codehafeez/cropimage/MainActivity.kt
package com.codehafeez.cropimage
import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.os.Environment
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.yalantis.ucrop.UCrop
import java.io.File
class MainActivity : AppCompatActivity() {

    private val PICK_IMAGE_REQUEST = 1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button: Button = findViewById(R.id.button)
        button.setOnClickListener {
            openImagePicker()
        }
    }

    private fun openImagePicker() {
        val intent = Intent(Intent.ACTION_PICK)
        intent.type = "image/*"
        startActivityForResult(intent, PICK_IMAGE_REQUEST)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (requestCode == PICK_IMAGE_REQUEST && resultCode == Activity.RESULT_OK) {
            val imageUri = data?.data
            imageUri?.let { startCropActivity(it) }
        } else if (requestCode == UCrop.REQUEST_CROP && resultCode == Activity.RESULT_OK) {
            val resultUri: Uri? = UCrop.getOutput(data!!)
        } else if (resultCode == UCrop.RESULT_ERROR) {
            val cropError: Throwable? = UCrop.getError(data!!)
        }
    }

    private fun startCropActivity(sourceUri: Uri) {
        val destinationUri: Uri = Uri.fromFile(createTempFile()) // Destination URI for the cropped image
        val maxWidth = 800 // Maximum width of the cropped image
        val maxHeight = 800 // Maximum height of the cropped image

        UCrop.of(sourceUri, destinationUri)
            .withAspectRatio(1f, 1f) // Set the aspect ratio (optional)
            .withMaxResultSize(maxWidth, maxHeight) // Set the maximum result size (optional)
            .start(this)
    }

    private fun createTempFile(): File {
        val tempFileName = "temp_cropped_image.jpg"
        val tempDir = getExternalFilesDir(Environment.DIRECTORY_PICTURES)
        return File(tempDir, tempFileName)
    }
}
